.github/workflows/lint.yml

```diff
@@ -0,0 +1,18 @@
+name: Lint
+on: [push, pull_request]
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.21
+      - name: Vet
+        run: go vet ./...
+      - name: Staticcheck
+        uses: dominikh/staticcheck-action@v1.3.0
```

Comment:
You should probably use a commit instead of a tag/branch to guarantee what code is being run on the CI. This is not really needed for GitHub's own actions though.